pandas/core/series.py

```diff
@@ -558,6 +559,39 @@ def _init_dict(
 
     # ----------------------------------------------------------------------
 
+    def __arrow_c_stream__(self, requested_schema=None):
+        """
+        Export the pandas Series as an Arrow C stream PyCapsule.
+
+        This relies on pyarrow to convert the pandas Series to the Arrow
+        format (and follows the default behaviour of ``pyarrow.Array.from_pandas``
+        in its handling of the index, i.e. to ignore it).
+        This conversion is not necessarily zero-copy.
+
+        Parameters
+        ----------
+        requested_schema : PyCapsule, default None
+            The schema to which the dataframe should be casted, passed as a
+            PyCapsule containing a C ArrowSchema representation of the
+            requested schema.
+
+        Returns
+        -------
+        PyCapsule
+        """
+        pa = import_optional_dependency("pyarrow", min_version="16.0.0")
+        if requested_schema is not None:
+            # todo: how should this be supported?
+            msg = (
+                "Passing `requested_schema` to `Series.__arrow_c_stream__` is not yet "
+                "supported"
+            )
+            raise NotImplementedError(msg)
```

Comment:
@kylebarron @jorisvandenbossche @WillAyd @PyCapsuleGang how should this be handled? I was looking at the Polars implementation and there's no tests there where `requested_schema` is not `None`

Reply:
In a general sense, you can ignore it.

> The callee should attempt to provide the data in the requested schema. However, if the callee cannot provide the data in the requested schema, they may return with the same schema as if None were passed to requested_schema.

https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html#schema-requests

However, in this case you should just delegate to pyarrow's implementation

```
        ca = pa.chunked_array([pa.Array.from_pandas(self, type=requested_schema)])
        return ca.__arrow_c_stream__(requested_schema)
```